### PR TITLE
Generate grant/revoke bnfs

### PR DESCRIPTION
**Pull request by jseldess**
- For granting privileges
- For granting roles
- For revoking privileges
- For revoking roles

### CODE REVIEW

**Comment by jseldess** [comment]
cc @lhirata

File: pkg/cmd/docgen/diagrams.go [hunk 1]
```diff
@@ -616,12 +616,29 @@ var specs = []stmtSpec{
...
+		},
 		unlink:  []string{"table_name", "database_name", "user_name"},
 		nosplit: true,
 	},
+	{
+		name: "grant_roles",
+		stmt: "grant_stmt",
+		replace: map[string]string{
+			"'GRANT' privileges 'ON' targets 'TO' name_list\n	| ": "",
+			"'GRANT' privilege_list 'TO' name_list": "'GRANT' ( ( ',' role_name ) )* 'TO' ( ( ',' user_name ) )*",
```
Comment (mberhault) [inline pkg/cmd/docgen/diagrams.go]: shouldn't this be `( ( role_name ) ( ( ',' role_name ) )*` (and same for `user_name`)? (or something similar, the parens look odd).

File: pkg/cmd/docgen/diagrams.go [hunk 2]
```diff
@@ -722,13 +739,29 @@ var specs = []stmtSpec{
...
+		},
+		unlink:  []string{"table_name", "database_name", "user_name"},
+		nosplit: true,
+	},
+	{
+		name: "revoke_roles",
+		stmt: "revoke_stmt",
+		replace: map[string]string{
+			"'REVOKE' privileges 'ON' targets 'FROM' name_list\n	| ": "",
+			"'REVOKE' privilege_list 'FROM' name_list": "'REVOKE' ( ( ',' role_name ) )* 'FROM' ( ( ',' user_name ) )*",
```
Comment (mberhault) [inline pkg/cmd/docgen/diagrams.go]: ditto here.

**Review by mberhault — APPROVED** [review]
Thanks for that.

File: pkg/cmd/docgen/diagrams.go (hunk 1 above)
Comment (jseldess) [reply pkg/cmd/docgen/diagrams.go]: Yes, I couldn't figure that out. Thank you!

File: pkg/cmd/docgen/diagrams.go (hunk 2 above)
Comment (jseldess) [reply pkg/cmd/docgen/diagrams.go]: Done.

**Comment by jseldess** [comment]
@lhirata, once this passes TC and merges, you should regenerate the diagrams from grant roles and revoke roles. With Marc's help, I got the bnfs right.

**Pull request merged**